app/controllers/devise_token_auth/concerns/set_user_by_token.rb

```diff
@@ -159,4 +163,10 @@ def auth_header_from_batch_request
     end
     auth_header
   end
+
+  def resource_is_checked?(mapping)
+    return true if @resource || eval("@has_checked_#{mapping}") == true
```

Comment:
Ugh, this is ugly (the `eval`), why do we need to have another instance variable?

Reply:
I think that is useful, When i have a many  `mapping` and not yet sign in, this instance variable to reject no need many DB connections.